packages/firestore/src/remote/watch_change.ts

```diff
@@ -350,35 +222,318 @@ export class WatchChangeAggregator {
     });
   }
 
+  /** Resets a target after an existence filter mismatch. */
+  handleExistenceFilterMismatch(targetId: TargetId): void {
+    this.resetTarget(targetId);
+  }
+
+  /**
+   * Converts the currently accumulated state into a remote event at the
+   * provided snapshot version. Resets the accumulated changes before returning.
+   */
+  createRemoteEvent(snapshotVersion: SnapshotVersion): RemoteEvent {
+    const targetChanges: { [targetId: number]: TargetChange } = {};
+
+    objUtils.forEachNumber(this.targetStates, (targetId, targetState) => {
+      if (this.isActiveTarget(targetId)) {
```

Comment:
This makes each RemoteEvent proportional to all active targets rather than just the targets with changes. This will create a ton of bookkeeping writes in the local store that we should avoid if possible.

Reply:
Addressed.

I am also garbage collection these states in TargetChange.REMOVE.